autosklearn/pipeline/components/regression/mlp.py

```diff
@@ -0,0 +1,239 @@
+                random_state=self.random_state,
+                verbose=self.verbose,
+                warm_start=True,
+                early_stopping=self.early_stopping_val,
+                validation_fraction=self.validation_fraction,
+                n_iter_no_change=self.n_iter_no_change,
+                tol=self.tol,
+                beta_1=self.beta_2,
+                beta_2=self.beta_1,
+                epsilon=self.epsilon,
+                # We do not use these, see comments below in search space
+                # momentum=self.momentum,
+                # nesterovs_momentum=self.nesterovs_momentum,
+                # power_t=self.power_t,
+                # learning_rate=self.learning_rate,
+                # max_fun=self.max_fun
+            )
+            self.scaler = sklearn.preprocessing.StandardScaler(copy=True)
+            self.scaler.fit(y.reshape((-1, 1)))
+        else:
+            new_max_iter = min(self.max_iter - self.estimator.n_iter_, n_iter)
+            self.estimator.max_iter = new_max_iter
+
+        Y_scaled = self.scaler.transform(y.reshape((-1, 1))).ravel()
+        self.estimator.fit(X, Y_scaled)
+        if self.estimator.n_iter_ >= self.max_iter or \
+                self.estimator._no_improvement_count > self.n_iter_no_change:
+            self._fully_fit = True
+        return self
+
+    def configuration_fully_fitted(self):
+        if self.estimator is None:
+            return False
+        elif not hasattr(self, '_fully_fit'):
+            return False
+        else:
+            return self._fully_fit
+
+    def predict(self, X):
+        if self.estimator is None:
+            raise NotImplementedError
+        Y_pred = self.estimator.predict(X)
+        return self.scaler.inverse_transform(Y_pred)
+
+    @staticmethod
+    def get_properties(dataset_properties=None):
+        return {'shortname': 'MLP',
+                'name': 'Multilayer Percepton',
+                'handles_regression': True,
+                'handles_classification': False,
+                'handles_multiclass': False,
+                'handles_multilabel': False,
+                'handles_multioutput': False,
+                'is_deterministic': True,
+                'input': (DENSE, SPARSE, UNSIGNED_DATA),
+                'output': (PREDICTIONS,)}
+
+    @staticmethod
+    def get_hyperparameter_search_space(dataset_properties=None):
+        cs = ConfigurationSpace()
+        hidden_layer_depth = UniformIntegerHyperparameter(name="hidden_layer_depth",
+                                                          lower=1, upper=3, default_value=1)
+        num_nodes_per_layer = UniformIntegerHyperparameter(name="num_nodes_per_layer",
+                                                           lower=16, upper=264, default_value=32,
+                                                           log=True)
+        activation = CategoricalHyperparameter(name="activation", choices=['tanh', 'logistic'],
```

Comment:
Do we really want to use the logistic activation function?

Reply:
That's what we decided some time ago. What would be better?